UnityProject/Assets/Scripts/Objects/Singularity.cs

```diff
@@ -234,7 +239,14 @@ private void PushObject(PushPull objectToPush, Vector3 pushVector)
 		{
 			if (CurrentStage == SingularityStages.Stage5 || CurrentStage == SingularityStages.Stage4)
 			{
-				objectToPush.GetComponent<RegisterPlayer>()?.ServerStun();
+				//Try stun player
+				if (DMMath.Prob(10) && TryGetComponent<PlayerHealth>(out var playerHealth) && playerHealth != null
+				&& !playerHealth.IsDead)
+				{
+					playerHealth.GetComponent<RegisterPlayer>()?.ServerStun();
```

Comment:
```suggestion
					playerHealth.GetComponent<RegisterPlayer>().OrNull()?.ServerStun();
```
Don't use null coalescing directly with Unity objects. Use the ``OrNull`` extension instead and then ``?``.